From fd4eb55519a5636f3264a6656321a2345b5958f1 Mon Sep 17 00:00:00 2001
From: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
Date: Mon, 26 Feb 2018 09:57:11 +0530
Subject: [PATCH 2/6] Fix for BLE pairing failure in slave role

Pairing fails due to DHkey mismatch and this fix address the
mismatch

Change-Id: Ie09f6c4ef3e70cce3f3b57858b6e8945eb65e63c
Tracked-On:https://jira01.devtools.intel.com/browse/OAM-57377
Signed-Off-by: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
---
 stack/btm/btm_ble_multi_adv.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/stack/btm/btm_ble_multi_adv.cc b/stack/btm/btm_ble_multi_adv.cc
index 99b8cb5..4085897 100644
--- a/stack/btm/btm_ble_multi_adv.cc
+++ b/stack/btm/btm_ble_multi_adv.cc
@@ -32,6 +32,7 @@
 #include "ble_advertiser.h"
 #include "ble_advertiser_hci_interface.h"
 #include "btm_int_types.h"
+#include "btm_int.h"
 
 using base::Bind;
 using base::TimeDelta;
@@ -202,7 +203,7 @@ class BleAdvertisingManagerImpl
     bda.address[5] = output.param_buf[0];
     bda.address[4] = output.param_buf[1];
     bda.address[3] = output.param_buf[2];
-
+    memcpy(&btm_cb.ble_ctr_cb.addr_mgnt_cb.private_addr,bda.address,6);
     cb.Run(bda);
   }
 
-- 
1.9.1

